packages/cozy-sharing/src/components/ShareByLink.jsx

```diff
@@ -1,22 +1,43 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import copy from 'copy-text-to-clipboard'
-import Toggle from 'cozy-ui/transpiled/react/Toggle'
-import { Spinner, SubTitle } from 'cozy-ui/transpiled/react'
 import Alerter from 'cozy-ui/transpiled/react/Alerter'
-import cx from 'classnames'
+import Text, { Caption } from 'cozy-ui/transpiled/react/Text'
+import Button from 'cozy-ui/transpiled/react/Button'
+import CompositeRow from 'cozy-ui/transpiled/react/CompositeRow'
+import Circle from 'cozy-ui/transpiled/react/Circle'
+import Icon from 'cozy-ui/transpiled/react/Icon'
+import Radio from 'cozy-ui/transpiled/react/Radio'
+import DropdownButton from 'cozy-ui/transpiled/react/DropdownButton'
+import ActionMenu, { ActionMenuItem } from 'cozy-ui/transpiled/react/ActionMenu'
+import { translate } from 'cozy-ui/transpiled/react/I18n'
+import get from 'lodash/get'
 import logger from '../logger'
 
-import styles from '../share.styl'
 import palette from 'cozy-ui/transpiled/react/palette'
 
+const checkWritePermissions = permissions => {
+  const permissionCategories = get(
+    permissions,
+    '[0].attributes.permissions',
+    {}
+  )
+  return (
+    Object.values(permissionCategories).filter(permissionCategory =>
+      get(permissionCategory, 'verbs', []).includes('POST')
+    ).length > 0
+  )
+}
+
```

Comment:
Since this API is not straightforward from a DX POV, we should add this kind of method to cozy-client directly. Either in the PermissionCollection or in a PermissionModel, no? 

I know we already have this kind of code in Notes, and we should mutualize.

Reply:
I updated the code to use the existing permission model, which did pretty much the same. I did keep the multi-permission check in here because I think this is specific to cozy-sharings.